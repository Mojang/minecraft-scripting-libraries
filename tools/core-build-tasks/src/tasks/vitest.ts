import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
