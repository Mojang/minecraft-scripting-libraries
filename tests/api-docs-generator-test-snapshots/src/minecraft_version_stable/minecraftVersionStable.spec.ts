// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Minecraft Version Stable', () => {
    it('Generates correct output for stable minecraft versions', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs', 'npm'],
            minecraftVersion: '1.2.3',
        });
    });
});
