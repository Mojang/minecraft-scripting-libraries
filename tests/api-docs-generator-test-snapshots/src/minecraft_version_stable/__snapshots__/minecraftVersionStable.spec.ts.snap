// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > # Of Files Generated 1`] = `13`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: minecraft/beta-module
    items:
    - name: minecraft/beta-module
      href: minecraft/beta-module/minecraft-beta-module.md
    - name: minecraft/beta-module Changelog
      href: minecraft/beta-module/changelog.md
  - name: minecraft/stable-module
    items:
    - name: minecraft/stable-module
      href: minecraft/stable-module/minecraft-stable-module.md
    - name: minecraft/stable-module Changelog
      href: minecraft/stable-module/changelog.md
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > msdocs/scriptapi/minecraft/beta-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/beta-module Changelog
description: Changelog of the \`@minecraft/beta-module\` module
---
# \`@minecraft/beta-module\` Changelog

## Version Changes
- [1.0.0-beta.1.2.3-stable](#100-beta123-stable)

## 1.0.0-beta.1.2.3-stable
#### Added \`@minecraft/beta-module\` Module
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > msdocs/scriptapi/minecraft/beta-module/minecraft-beta-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/beta-module Module
description: Contents of the @minecraft/beta-module module
monikerRange: "=minecraft-bedrock-experimental"
---
# \`@minecraft/beta-module\` Module

> [!CAUTION]
> This module is still in pre-release.  It may change or it may be removed in future releases.

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "@minecraft/beta-module",
    "version": "1.0.0-beta.1.2.3-stable"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0-beta.1.2.3-stable\`
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > msdocs/scriptapi/minecraft/stable-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/stable-module Changelog
description: Changelog of the \`@minecraft/stable-module\` module
---
# \`@minecraft/stable-module\` Changelog

## Version Changes
- [1.0.0](#100)

## 1.0.0
#### Added \`@minecraft/stable-module\` Module
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > msdocs/scriptapi/minecraft/stable-module/minecraft-stable-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: minecraft/stable-module Module
description: Contents of the @minecraft/stable-module module
---
# \`@minecraft/stable-module\` Module

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "@minecraft/stable-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/beta-module@1.0.0-beta.1.2.3-stable/README.md 1`] = `
"# \`@minecraft/beta-module\`

## **NOTE: This version of this module is still in pre-release.  It may change or it may be removed in future releases.**

See full documentation for this module here:

https://learn.microsoft.com/en-us/minecraft/creator/scriptapi/minecraft/beta-module/minecraft-beta-module"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/beta-module@1.0.0-beta.1.2.3-stable/index.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "@minecraft/beta-module",
 *   "version": "1.0.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/beta-module@1.0.0-beta.1.2.3-stable/package.json 1`] = `
"{
  "name": "@minecraft/beta-module",
  "version": "1.0.0-beta.1.2.3-stable",
  "description": "",
  "contributors": [
    {
      "name": "Jake Shirley",
      "email": "jake@xbox.com"
    },
    {
      "name": "Mike Ammerlaan",
      "email": "mikeam@microsoft.com"
    }
  ],
  "license": "MIT"
}"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/stable-module@1.0.0/README.md 1`] = `
"# \`@minecraft/stable-module\`

See full documentation for this module here:

https://learn.microsoft.com/en-us/minecraft/creator/scriptapi/minecraft/stable-module/minecraft-stable-module"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/stable-module@1.0.0/index.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "@minecraft/stable-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > npm/types/@minecraft/stable-module@1.0.0/package.json 1`] = `
"{
  "name": "@minecraft/stable-module",
  "version": "1.0.0",
  "description": "",
  "contributors": [
    {
      "name": "Jake Shirley",
      "email": "jake@xbox.com"
    },
    {
      "name": "Mike Ammerlaan",
      "email": "mikeam@microsoft.com"
    }
  ],
  "license": "MIT"
}"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > typescript/@minecraft/beta-module@1.0.0-beta.1.2.3-stable.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "@minecraft/beta-module",
 *   "version": "1.0.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Minecraft Version Stable > Generates correct output for stable minecraft versions > typescript/@minecraft/stable-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "@minecraft/stable-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
"
`;
